Provide consistent examples of bindings for all SDKs

amulyavarote opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Modified bindings documentation page (under building blocks) for dapr.io

## Issue reference

This partially fulfills: #1969

berndverst committed:
Fix incorrect warning log-level

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

berndverst committed:
Correct the default component init timeout

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Added pub sub documentation

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Changed port number in the command

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Modified based on the review comments - 1

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Modified based on the review comments - 2

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Added bindings documentation

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Changed python commands

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Changed python commands

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Changed python commands

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:
```diff
@@ -9,19 +9,19 @@ weight: 300
-<div class="embed-responsive embed-responsive-16by9">
-<iframe width="560" height="315" src="https://www.youtube.com/embed/ysklxm81MTs?start=1960" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources(output binding).
```
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources via an output binding.
```

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:
```diff
@@ -9,19 +9,19 @@ weight: 300
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources(output binding).
+
+<img src="/images/building-block-bindings-example.png" width=1000 alt="Diagram showing bindings of example service">
```
this image has an extra thin grey line at the right hand side that needs to be cropped out.

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
+
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application(input binding).
+
+<img src="/images/building-block-bindings-example.png" width=1000 alt="Diagram showing bindings of example service">
```
this image has an extra thin grey line at the right hand side that needs to be cropped out.

Commit:
Modified based on the review comments - 3

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Changed image for bindings

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
 ## 1. Create a binding
 
-An input binding represents an event resource that Dapr uses to read events from and push to your application.
+An output binding represents a resource that Dapr uses to invoke and send messages to.
```
should this be describing an input biding instead?

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
-For a quickstart sample showing bindings, visit this [link](https://github.com/dapr/quickstarts/tree/master/bindings).
+## Example:
+
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application(input binding).
```
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application via an input binding.
```

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
+{{< /tabs >}}
 
-## 2. Listen for incoming events
+## 2. Listen for incoming events(Input Binding)
```
```suggestion
## 2. Listen for incoming events (input binding)
```

willtsai commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
+
+### Rejecting an event
+
+In order to tell Dapr that the event wasn't processed correctly in your application and schedule it for redelivery, return any response different from `200 OK`. For example, a `500 Error`.
```
```suggestion
In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery, return any response other than `200 OK`. For example, a `500 Error`.
```

Commit:
Modified based on the review comments - 1

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:
```diff
@@ -9,19 +9,19 @@ weight: 300
 
 An output binding represents a resource that Dapr uses to invoke and send messages to.
 
 For the purpose of this guide, you'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref setup-bindings >}}).
```
```suggestion
For the purpose of this guide, you'll use a Kafka binding. You can find a list of all supported binding components [here]({{< ref setup-bindings >}}).
```

Commit:
Modified based on the review comments - 2

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Reverted a small change

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

dlm committed:
Fix publish a topic examples

Step 3 and 4 of the "How-To: Publish & subscribe" publishes to the
wrong app id and pubsub. This commit fixes by changing the examples
that write to the app-id "testpubsub" to "orderprocessing"
and the pubsub "pubsub" to "order_pub_sub"

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

greenie-msft committed:
Wrap oauth links in ignore comment

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

greenie-msft committed:
Update dapr versions for 1.5.1 hotfix

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

emctl committed:
Bumping tags to v1.5.0

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
+Now configure your application to receive incoming events. If using HTTP, you need to listen on a `POST` endpoint with the name of the binding as specified in `metadata.name` in the file. 
 
-*The following example shows how you would listen for the event in Node.js, but this is applicable to any programming language*
+Below are code examples that leverage Dapr SDKs for output binding.
```
```suggestion
Below are code examples that leverage Dapr SDKs to demonstrate an output binding.
```

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
+
+### Rejecting an event
+
+In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery, return any response other than `200 OK`. For example, a `500 Error`.
```
Do we really want to encourage returning a 500 error? Shouldn't this be handled gracefully?

amulyavarote replied:
This part was present in the docs perviously. I can take it out if not needed.

Commit:
Modified based on the review comments - 2

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:
```diff
@@ -9,19 +9,19 @@ weight: 300
-<div class="embed-responsive embed-responsive-16by9">
-<iframe width="560" height="315" src="https://www.youtube.com/embed/ysklxm81MTs?start=1960" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources via an output binding.
```
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources, in this case a Kafka, via an output binding.
```

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:
```diff
@@ -59,38 +67,271 @@ To deploy this into a Kubernetes cluster, fill in the `metadata` connection deta
-## 2. Send an event
+## 2. Send an event(Output Binding)
+
+Below are code examples that leverage Dapr SDKs for output binding.
```
```suggestion
Below are code examples that leverage Dapr SDKs to demonstrate an output binding.
```

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
+An input binding represents a resource that Dapr uses to read events from and push to your application.
 
-For the purpose of this HowTo, we'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref supported-bindings >}}).
+For the purpose of this guide, you'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref setup-bindings >}}).
```
```suggestion
For the purpose of this guide, you'll use a Kafka binding. You can find a full list of supported binding components [here]({{< ref setup-bindings >}}).
```

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
-(Use the `--components-path` flag with `dapr run` to point to your custom components dir)
+Create a new binding component with the name of `checkout`.
+
+Inside the `metadata` section, configure Kafka related properties such as the topic to publish the message to and the broker.
```
```suggestion
Inside the `metadata` section, configure Kafka related properties, such as the topic to publish the message to and the broker.
```

greenie-msft commented on daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
+
+### Rejecting an event
+
+In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery and handle it gracefully using logs or exceptions.
```
Lets put this back to how it was until we discuss how best this should be handled. 

```suggestion
In order to tell Dapr that the event wasn't processed correctly in your application and schedule it for redelivery, return any response different from `200 OK`. For example, a `500 Error`.
```

Commit:
Modified based on the review comments -3

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

willtsai commented:
Looks like this comment from me is not yet addressed. Please either accept and commit the suggestion, or else resolve the comment if you are taking this in another direction.
https://github.com/dapr/docs/pull/2021/files#r762207024

Commit:
Modified a line based on the review comment

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Modified a line based on the review comment

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>

Commit:
Modified based on the review comments

Signed-off-by: Amulya Varote <amulyavarote@MININT-8V4CL0S.redmond.corp.microsoft.com>
